mods/gen1/random-teams.js

```diff
@@ -274,8 +273,8 @@ class RandomGen1Teams extends RandomGen2Teams {
 			if (movePool.length) {
 				hasMove = {};
 				counter = {Physical: 0, Special: 0, Status: 0, physicalsetup: 0, specialsetup: 0};
-				for (let k = 0; k < moves.length; k++) {
-					let move = this.getMove(moves[k]);
+				for (const i of moves) {
```

Comment:
I'd use `id` here

Reply:
In hindsight, all the places I've been saying `id`, we might want to use `setMoveid`

The main difference between `moveid` and `setMoveid` would be that `setMoveid` includes Hidden Power type information.